libdnf/module/PlatformModulePackage.cpp

```diff
@@ -0,0 +1,70 @@
+#include "PlatformModulePackage.hpp"
+#include "libdnf/utils/File.hpp"
+
+#include <algorithm>
+#include <sstream>
+
+Id PlatformModulePackage::createSolvable(Pool *pool, HyRepo repo, const std::string &osReleasePath,
+                                         const std::string &arch)
+{
+    Id id = repo_add_solvable(repo->libsolv_repo);
+    Solvable *solvable = pool_id2solvable(pool, id);
+
+    std::string name = "platform";
+    std::string stream = getPlatformStream(osReleasePath);
+    long version = 1;
+    std::string context = "0";
+
+    std::ostringstream ss;
+    ss << name << "-" << stream << "-" << version << "-" << context;
+    solvable_set_str(solvable, SOLVABLE_NAME, ss.str().c_str());
+    solvable_set_str(solvable, SOLVABLE_EVR, "0");
+    solvable_set_str(solvable, SOLVABLE_ARCH, arch.c_str());
+
+    ss.str(std::string());
+    ss << "module-n(" << name << ")";
+    auto depId = pool_str2id(pool, ss.str().c_str(), 1);
+    solvable_add_deparray(solvable, SOLVABLE_PROVIDES, depId, -1);
+
+    ss.str(std::string());
+    ss << "module-ns(" << name << ":" << stream << ")";
+    depId = pool_str2id(pool, ss.str().c_str(), 1);
+    solvable_add_deparray(solvable, SOLVABLE_PROVIDES, depId, -1);
+
+    ss.str(std::string());
+    ss << "module-nsv(" << name << ":" << stream << ":" << version << ")";
+    depId = pool_str2id(pool, ss.str().c_str(), 1);
+    solvable_add_deparray(solvable, SOLVABLE_PROVIDES, depId, -1);
+
+    return id;
+}
+
+PlatformModulePackage::PlatformModulePackage(Pool *pool, HyRepo repo, const std::string &osReleasePath,
+                                             const std::string &arch)
+{
+    id = createSolvable(pool, repo, osReleasePath, arch);
+}
+
+std::string PlatformModulePackage::getPlatformStream(const std::string &osReleasePath)
+{
+    auto file = libdnf::File::newFile(osReleasePath);
+    std::string line;
+    while (file->readLine(line)) {
```

Comment:
isn't there already ini-format reader in the code?

Reply:
@ignatenkobrain /etc/os-release is not an ini file, see man os-release:

> The basic file format of os-release is a newline-separated list of environment-like
> shell-compatible variable assignments. It is possible to source the configuration
> from shell scripts ...

Reply:
https://github.com/rpm-software-management/libdnf/blob/master/libdnf/dnf-context.cpp#L1105-L1156